### PR TITLE
Added reference-store to snippet data-provider

### LINKED ISSUE #3439: Use SnippetReferenceStore in SnippetDataProvider
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6
| Browser Version | --

#### Actual Behavior

SnippetReferenceStore isn't used in SnippetDataProvider.

#### Expected Behavior

Snippet references are also stored via SmartContent.

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #3439
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR introduces the `exclude_duplicates` configuration to `SnippetDataProvider`.

#### Why?

This is missing in the data-provider.

### CODE REVIEW

**Comment by danrot** [comment]
The tests are failing

**Comment by wachterjohannes** [comment]
@danrot tests should go green now

**Comment by danrot** [comment]
They don't 😛

File: src/Sulu/Bundle/SnippetBundle/Tests/Functional/Import/SnippetImportTest.php [hunk 1]
```diff
@@ -25,20 +29,35 @@ class SnippetImportTest extends SuluTestCase
...
+     */
+    protected $path;
 
     /**
      * Setup data for import.
      */
     protected function setUp()
     {
+        $this->distPath = __DIR__ . '/../../app/Resources/import/export.xliff.dist';
+        $this->path = __DIR__ . '/../../app/Resources/import/export.xliff';
```
Comment (danrot) [inline src/Sulu/Bundle/SnippetBundle/Tests/Functional/Import/SnippetImportTest.php]: Was this one of the failing tests? Why does it have to be done like this?

File: src/Sulu/Bundle/SnippetBundle/Content/SnippetDataProvider.php [hunk 2]
```diff
@@ -265,11 +256,22 @@ private function resolveFilters(array $filters, array $propertyParameter, array
...
             ? $propertyParameter['properties']->getValue() : [];
 
+        $excluded = [];
+
+        if (array_key_exists('excluded', $filters) && is_array($filters['excluded'])) {
+            $excluded = $filters['excluded'];
+        }
+
+        if (array_key_exists('exclude_duplicates', $propertyParameter)
+            && $propertyParameter['exclude_duplicates']->getValue()) {
```
Comment (danrot) [inline src/Sulu/Bundle/SnippetBundle/Content/SnippetDataProvider.php]: Don't we usually write the last `) {` into a separate line?

**Pull request merged**